packages/xo-server/src/api/vm.js

```diff
@@ -746,7 +746,9 @@ export async function snapshot ({
 }) {
   await checkPermissionOnSrs.call(this, vm)
 
-  return (await this.getXapi(vm).snapshotVm(vm._xapiRef, name)).$id
+  const snapshotId = (await this.getXapi(vm).snapshotVm(vm._xapiRef, name)).$id
+  await this.addAcl(this.user.id, snapshotId, 'admin')
```

Comment:
Should we do that all the time or only when the snapshot has been created via ACLs?

Reply:
I think, if possible (not too difficult), we should not create unnecessary ACLs, only if the snapshot has been created thanks to ACLs.